tools/pdffontsinfo.py

```diff
@@ -0,0 +1,222 @@
+    parser.add_argument('-P', '--password')
+    parser.add_argument('-o', '--output')
+    parser.add_argument('-R', '--rotation')
+    parser.add_argument('-p', '--pagenos')
+    parser.add_argument('-m', '--maxpages')
+    parser.add_argument('-C', '--disable-caching', action='store_true')
+    parser.add_argument('-n', '--no-layout', action='store_true')
+    parser.add_argument('-A', '--all-texts', action='store_true')
+    parser.add_argument('-V', '--detect-vertical', action='store_true')
+    parser.add_argument('-M', '--char-margin')
+    parser.add_argument('-L', '--line-margin')
+    parser.add_argument('-W', '--word-margin')
+    parser.add_argument('-F', '--boxes-flow')
+    parser.add_argument('-d', '--debug', action='store_true')
+    args = parser.parse_args()
+
+    # debug option
+    debug = 0
+    # input option
+    password = b''
+    pagenos = set()
+    maxpages = 0
+    # output option
+    outfile = None
+    rotation = 0
+    caching = True
+
+    laparams = LAParams()
+    if args.debug:
+        debug += 1
+    elif args.password:
+        password = args.password.encode('ascii')
+    elif args.output:
+        outfile = args.output
+    elif args.rotation:
+        rotation = int(args.rotation)
+    elif args.pagenos:
+        pagenos.update(int(x)-1 for x in args.pagenos.split(','))
+    elif args.maxpages:
+        maxpages = int(args.maxpages)
+    elif args.disable_caching:
+        caching = False
+    elif args.no_layout:
+        laparams = None
+    elif args.all_texts:
+        laparams.all_texts = True
+    elif args.detect_vertical:
+        laparams.detect_vertical = True
+    elif args.char_margin:
+        laparams.char_margin = float(args.char_margin)
+    elif args.word_margin:
+        laparams.word_margin = float(args.word_margin)
+    elif args.line_margin:
+        laparams.line_margin = float(args.line_margin)
+    elif args.boxes_flow:
+        laparams.boxes_flow = float(args.boxes_flow)
+
+    #
+    PDFDocument.debug = debug
+    PDFParser.debug = debug
+    CMapDB.debug = debug
+    PDFPageInterpreter.debug = debug
+    #
+
+    rsrcmgr = PDFResourceManager(caching=caching)
+
+    if outfile:
+        outfp = open(outfile, 'w')
+    else:
+        outfp = sys.stdout
+
+    device = PDFConverter(
+        rsrcmgr, outfp, laparams=laparams)
+
+    for fname in args.input:
+        with open(fname, 'rb') as fp:
+            interpreter = PDFPageInterpreter(rsrcmgr, device)
+            for page in PDFPage.get_pages(
+                fp,
+                pagenos,
+                maxpages=maxpages,
+                password=password,
+                caching=caching,
+                check_extractable=True,
+            ):
+                page.rotate = (page.rotate + rotation) % 360
+                interpreter.process_page(page)
+
+        l_font_names = []
+        l_font_types = []
+        l_font_enc = []
+        l_font_uni = []
```

Comment:
You can remove these and use the fonts object directly

Reply:
Fixed! Thanks for the suggestion.